pkg/client/clientset_generated/internalclientset/typed/policy/internalversion/eviction_expansion.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internalversion
+
+import (
+	"strings"
+
+	policy "k8s.io/kubernetes/pkg/apis/policy"
+)
+
+// The EvictionExpansion interface allows manually adding extra methods to the ScaleInterface.
+type EvictionExpansion interface {
+	Evict(eviction *policy.Eviction) error
+}
+
+func (c *evictions) Evict(eviction *policy.Eviction) error {
```

Comment:
@caesarxuchao Correct me if I'm wrong.
Based on our offline discussion with @caesarxuchao , we have thought about using a similar approach as `scale` subresource. But what we found is no one is actually calling [Update()](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/internalclientset/typed/extensions/internalversion/scale_expansion.go#L49) to post `scale` subresource.
And I guess if we try to call this function to post a `scale`, we will get [a similar error as eviction](https://github.com/kubernetes/kubernetes/issues/36305).